File: java/src/main/java/com/sasteval/vuln/hard/QueryBuilder.java
package com.sasteval.vuln.hard;

/**
 * CWE-89: SQL Injection (Cross-File Helper - Query Builder)
 * Part of a 3-file taint chain: SqlInjectionCrossFile -> QueryBuilder -> QueryExecutor.
 */
public class QueryBuilder {

    public String buildUserQuery(String userId) {
        // VULN: User input concatenated into SQL query string
        return "SELECT * FROM users WHERE id = '" + userId + "'";
    }
}

File: dotnet/Vuln/Hard/QueryBuilder.cs
namespace SastEval.Vuln.Hard;

/// <summary>
/// Helper class for SqlInjectionCrossFile - builds SQL query with string concatenation.
/// </summary>
public static class QueryBuilder
{
    public static string BuildUserQuery(string userId)
    {
        // VULNERABLE: User input concatenated into SQL query string
        return "SELECT * FROM users WHERE id = '" + userId + "'";
    }
}
